### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by dimak911**
handle user profile

### CODE REVIEW

File: src/auth/local.strategy.ts [hunk 1]
```diff
@@ -11,7 +12,10 @@ export class LocalStrategy extends PassportStrategy(Strategy) {
     });
   }
 
-  public async validate(email: string, password: string) {
+  public async validate(
+    email: string,
+    password: string
+  ): Promise<Omit<User, 'password'>> {
```
Comment (vadymsushkovdev) [inline src/auth/local.strategy.ts]: Type "Omit<User, 'password'>" is duplicating. Make special type for that and use it

File: src/profile/entities/profile.entity.ts [hunk 2]
```diff
@@ -1,31 +1,21 @@
...
-  @Column({ name: 'last_name', nullable: true })
+  @Column({ name: 'last_name', default: '' })
   lastName: string;
 
-  @Column({ name: 'avatar', nullable: true })
+  @Column({ name: 'avatar', default: '' })
   avatar: string;
 
-  @Column({ name: 'description', nullable: true })
+  @Column({ name: 'description', default: '' })
```
Comment (vadymsushkovdev) [inline src/profile/entities/profile.entity.ts]: Pls, change "default: true" on "nullable: true"

File: src/profile/profile.controller.ts [hunk 3]
```diff
@@ -1,49 +1,45 @@
...
   constructor(private readonly profileService: ProfileService) {}
 
-  @Post()
-  public create(@Body() createProfileDto: CreateProfileDto) {
-    return this.profileService.create(createProfileDto);
+  @Get(':id')
+  public findOneById(
+    @Param() { id }: IdParams
+  ): Promise<ProfileResponseDto> {
+    return this.profileService.findOneById(+id);
```
Comment (vadymsushkovdev) [inline src/profile/profile.controller.ts]: Use @Transform method from library "class-transformer" to make string parameter to number

Comment (vadymsushkovdev) [inline src/profile/profile.controller.ts]: Use @Transform method from library "class-transformer" to make string parameter to number

File: src/user/user.service.ts [hunk 4]
```diff
@@ -5,8 +5,9 @@ import { Repository } from 'typeorm';
 import { JwtService } from '@nestjs/jwt';
 import { User } from '@src/user/entities/user.entity';
 import { ProfileService } from '@src/profile/profile.service';
-import { IUser } from '@src/user/models/user.interface';
 import { MailService } from '@src/mail/mail.service';
+import * as bcrypt from 'bcrypt';
```
Comment (vadymsushkovdev) [inline src/user/user.service.ts]: "import *" is bad case to use. Try a destructurization case like "import {hash, compare} from 'bcrypt'"

File: src/auth/local.strategy.ts (hunk 1 above)
Comment (dimak911) [reply src/auth/local.strategy.ts]: done

File: src/profile/entities/profile.entity.ts (hunk 2 above)
Comment (dimak911) [reply src/profile/entities/profile.entity.ts]: done

File: src/user/user.service.ts (hunk 4 above)
Comment (dimak911) [reply src/user/user.service.ts]: done

File: src/profile/profile.controller.ts (hunk 3 above)
Comment (dimak911) [reply src/profile/profile.controller.ts]: done

**Review by vadymsushkovdev — APPROVED** [review]

**Pull request closed without merging**